### PR TITLE
v17.2.2

### PR DESCRIPTION
**Pull request by djgalloway**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by djgalloway** [comment]
The version commit's just going to have to go unsigned.  I meant to https://github.com/ceph/ceph-build/pull/2026 first but it got forgotten.

File: qa/tasks/cephfs/test_volumes.py [hunk 1]
```diff
@@ -6330,3 +6330,143 @@ def test_subvolume_upgrade_v1_to_v2(self):
...
         self._wait_for_trash_empty()
+
+    def test_malicious_metafile_on_legacy_to_v1_upgrade(self):
+        """
+        Validate handcrafted .meta file on legacy subvol root doesn't break the system
+        on legacy subvol upgrade to v1
+        poor man's upgrade test -- theme continues...
+        """
+        subvol1, subvol2 = self._generate_random_subvolume_name(2)
+        group = self._generate_random_group_name()
```
Comment (idryomov) [inline qa/tasks/cephfs/test_volumes.py:6341]: @kotreshhr This variable is assigned but not used, which breaks `run-tox-qa` test.

File: debian/changelog [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+
+  * New upstream release
+
+ -- Ceph Release Team <ceph-maintainers@ceph.com>  Thu, 21 Jul 2022 17:29:31 +0000
+
+ceph (17.2.1-1) stable; urgency=medium
+
+  * New upstream release
+
+ -- Ceph Release Team <ceph-maintainers@ceph.com>  Thu, 23 Jun 2022 14:41:32 +0000
```
Comment (idryomov) [inline debian/changelog:11]: @djgalloway This PR includes "tools: ceph-objectstore-tool is able to trim pg log dups' entries." and "17.2.1" commits which are already in quincy (merged via https://github.com/ceph/ceph/pull/46827).

**Review by idryomov — CHANGES_REQUESTED** [review]

**Comment by idryomov** [comment]
@kotreshhr We also need to get these changes into the main branch -- please open a PR against main after fixing the test.

File: qa/tasks/cephfs/test_volumes.py (hunk 1 above)
Comment (kotreshhr) [reply qa/tasks/cephfs/test_volumes.py:6341]: @idryomov my bad. @djgalloway would you fix this ? I can do that. Do I have permission to push to your branch ?

**Comment by kotreshhr** [comment]
> @kotreshhr We also need to get these changes into the main branch -- please open a PR against main after fixing the test.

Sure, I will do that.

**Comment by kotreshhr** [comment]
> > @kotreshhr We also need to get these changes into the main branch -- please open a PR against main after fixing the test.
> 
> Sure, I will do that.

Done. https://github.com/ceph/ceph/pull/47229

File: qa/tasks/cephfs/test_volumes.py (hunk 1 above)
Comment (djgalloway) [reply qa/tasks/cephfs/test_volumes.py:6341]: Yes, of course.

File: debian/changelog (hunk 2 above)
Comment (djgalloway) [reply debian/changelog:11]: Yeah, I thought those would disappear out of this PR once https://github.com/ceph/ceph/pull/46827 was merged.  I'm not sure how the commits are broken.

File: qa/tasks/cephfs/test_volumes.py (hunk 1 above)
Comment (djgalloway) [reply qa/tasks/cephfs/test_volumes.py:6341]: @kotreshhr you have permission to push to this branch.  I don't know how to fix the test.

File: debian/changelog (hunk 2 above)
Comment (idryomov) [reply debian/changelog:11]: I think you need to rebase on top of commit f61704e9a484e828fc16639456e5747f3e54e768.

Comment (idryomov) [reply debian/changelog:11]: No, scratch that -- I wasn't thinking straight.

File: qa/tasks/cephfs/test_volumes.py (hunk 1 above)
Comment (djgalloway) [reply qa/tasks/cephfs/test_volumes.py:6341]: Are you suggesting just cherry-picking https://github.com/ceph/ceph/pull/47229/commits/c8416c3f0e15e49dde4b9bca6f732231d9b5d16b now?

Comment (idryomov) [reply qa/tasks/cephfs/test_volumes.py:6341]: Yes, but preferably after https://github.com/ceph/ceph/pull/47229 is merged.

**Review by idryomov — APPROVED** [review]
LGTM now (although I didn't expect that the tag would get recreated -- the fixup could have been just rebased in...)

**Comment by djgalloway** [comment]
Cherry-picking that commit and reordering changed the version sha.  I don't think it matters though?

**Comment by idryomov** [comment]
I expected it to be cherry picked on top of the tag, as the tag already "went out".

**Comment by djgalloway** [comment]
> I expected it to be cherry picked on top of the tag, as the tag already "went out".

Good point.  Will fix.

**Pull request merged**